### PR TITLE
Cleanup: dev, demo and linting

### PR DESCRIPTION
**Pull request by qazwsxpawel**
Started by trying to make _clj-kondo_ a bit happier :)

* some mechanical changes 
* few less mechanical (should double check, but seems everything is still in working order ;p)
* I've pulled `test` to top top level
* I've added a demo ns (which we can remove after this week or move and develop if that's useful)

### CODE REVIEW

File: src/clojuratica/base/parse.clj [hunk 1]
```diff
@@ -94,9 +94,9 @@
 
 (defn bound-map [f coll]
   (cond
-    (options/flag? dynamic-vars/*options* :vectors) (vec-bound-map f coll)
-    (options/flag? dynamic-vars/*options* :seqs     (seq-bound-map f coll))
-    (options/flag? dynamic-vars/*options* :seq-fn   (seq-fn-bound-map f coll))))
+    (options/flag? dynamic-vars/*options* :vectors)  (vec-bound-map f coll)
+    (options/flag? dynamic-vars/*options* :seqs)     (seq-bound-map f coll)
```
Comment (qazwsxpawel) [inline src/clojuratica/base/parse.clj:98]: This looked to me like parens in a wrong place

File: src/clojuratica/base/parse.clj [hunk 2]
```diff
@@ -163,16 +163,16 @@
...
 (defn parse-hash-map [expr]
   (debug/with-debug-message (options/flag? dynamic-vars/*options* :verbose) "hash-map parse"
     (let [inside    (first (.args expr))
           rules     (parse
                       (cond (.listQ inside) inside
                             (= "Dispatch" (expr/head-str inside)) (first (.args inside))
-                            'else (assert (or (.listQ inside))
-                                          (= "Dispatch" (expr/head-str inside)))))
+                            :else (assert (or (.listQ inside)
+                                              (= "Dispatch" (expr/head-str inside))))))
```
Comment (qazwsxpawel) [inline src/clojuratica/base/parse.clj:175]: Again, this looks like the `=` form should be inside `or`

File: src/clojuratica/base/evaluate.clj [hunk 3]
```diff
@@ -1,8 +1,7 @@
 (ns clojuratica.base.evaluate
-  (:require [clojuratica.lib.debug :as debug]
-            [clojuratica.lib.options :as options]
+  (:require [clojuratica.lib.options :as options]
             [clojuratica.base.convert :as convert]
-            [clojuratica.runtime.default-options :as default-options]
+            #_[clojuratica.runtime.default-options :as default-options]
```
Comment (reborg) [inline src/clojuratica/base/evaluate.clj:4]: Are there reasons to keep it around?

File: src/clojuratica/base/parse.clj (hunk 1 above)
Comment (reborg) [reply src/clojuratica/base/parse.clj:98]: Seems likely, good catch...

**Review by reborg — APPROVED** [review]
All legit to me!

**Review by metasoarous — APPROVED** [review]
Looks good! Gonna merge these so I can try things out!

**Pull request merged**